main.tf

```diff
@@ -1,42 +1,48 @@
 locals {
   enabled = module.this.enabled
+
+  lambda_src = "${path.module}/lambda"
 }
 
-resource "random_id" "build_id" {
+resource "random_id" "build" {
   count = local.enabled ? 1 : 0
 
   byte_length = 8
   keepers = {
-    "hash" = filesha256("${path.module}/lambda/main.py")
+    hash = filesha256("${local.lambda_src}/main.py")
   }
 }
 
 data "archive_file" "build" {
   count = local.enabled ? 1 : 0
 
-  source_dir  = "${path.module}/lambda"
-  output_path = "/tmp/${random_id.build_id[0].hex}.zip"
+  source_dir  = local.lambda_src
+  output_path = "/tmp/${random_id.build[0].hex}.zip"
   type        = "zip"
 }
 
 module "lambda" {
-  source = "git@github.com:rallyware/terraform-aws-lambda-function.git"
+  count = local.enabled ? 1 : 0
```

Comment:
not required, submodule works with context module where enabled flag already provided